todoapp/src/components/AddTask.tsx

```diff
@@ -0,0 +1,68 @@
+//**** Instead this Added New Component(CreateTask), this is only for Future Ref. Purpose */
+
+import { useState, FormEvent } from "react";
+import { useNavigate } from "react-router-dom";
+import { GenerateRandomNo } from "../helper/helper";
+
+const AddTask = () => {
+  const [title, setTitle] = useState("");
+  const [dueDate, setDueDate] = useState("");
+  const navigate = useNavigate();
+
+  const addTodo = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const todoItem = {
+      id: GenerateRandomNo(),
+      task: title,
+      isComplete: false,
+      dueDate: dueDate,
+    };
+
+    fetch(process.env.REACT_APP_BASE_URL + "/todos", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(todoItem),
+    }).then(() => {
+      setTitle("");
+      setDueDate("");
+      navigate("/");
+    });
+  };
+
+  return (
+    <div className="form-details text-center">
+      <form onSubmit={addTodo}>
+        <h2>- Add Task - </h2>
+        <label htmlFor="title">Title : </label>
+        <input
+          type="text"
+          name="title"
+          id="title"
+          value={title}
+          onChange={(e) => setTitle(e.target.value)}
```

Comment:
inline function should be only in case of dynamic stuff

Reply:
Resolved !!